examples/echo-server/src/main.rs

```diff
@@ -27,6 +28,27 @@ async fn main() -> Result<ExitCode, Error> {
         eprintln!("Error: {}", error);
     }));
 
+    // Map the request and response bodies to a boxed body type with 2 levels
+    // of indirection. Also, connect a sink a sync to the request and response
+    // body for the purpose of audit logs (because enterprise software).
+    app.include(|request: Request, next: Next| {
```

Comment:
As you can see here, by removing the tee method and a couple of helper functions to `HttpBody`, we can perform a common task required for many enterprise-grade applications for both the sake of information security and audit logs without having to connect the sink to the actual socket. This approach prevents somebodies JWT and / or session cookie from being stolen by someone with access to the hypothetical audit logs.

We definitely need more documentation on the body module. Including examples that use `http-body-util`. I can see my self doing a couple of transformations with the `BodyExt` trait. Such as calling `flat_map` on `TeeBody` to compress the bytes that are written to the socket without affecting the data that is written to the sink. Lot's of potential here.

Reply:
Also this example doesn't work because the echo server already uses a boxed body. I'm going to update the example but share a code snippet below because it's valid for any other use case that exclusively uses `ResponseBody`.

```
app.include(|request: Request, next: Next| {
        let future = next.call(request.map(|existing| {
            BoxBody::new(TeeBody::new(
                BoxBody::new(existing.into_inner()),
                tokio::io::stderr(),
            ))
        }));

        async {
            Ok(future.await?.map(|existing| {
                BoxBody::new(TeeBody::new(
                    BoxBody::new(existing.into_inner()),
                    tokio::io::stdout(),
                ))
            }))
        }
});
```

Reply:
On second thought, I like the updated example better with the reintroduction of `TeeBody::cap`.